kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableWritesSuite.scala

```diff
@@ -1009,4 +1004,228 @@ class DeltaTableWritesSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBa
+      assert(table.getLatestSnapshot(engine).getSchema(engine).equals(testSchema))
+    }
+  }
+
+  test("update table with unsupported column mapping mode") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      createTxn(engine, tablePath, isNewTable = true, testSchema, Seq.empty)
+        .commit(engine, emptyIterable())
+
+      val ex = intercept[InvalidConfigurationValueException] {
+        table.createTransactionBuilder(engine, testEngineInfo, Operation.WRITE)
+          .withTableProperties(
+            engine,
+            Map(ColumnMapping.COLUMN_MAPPING_MODE_KEY -> "invalid").asJava)
+          .build(engine)
+      }
+      assert(ex.getMessage.contains("Invalid value for table property " +
+        "'delta.columnMapping.mode': 'invalid'. Needs to be one of none, id, name."))
+    }
+  }
+
+  test("update table with unsupported column mapping mode change") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      createTxn(engine, tablePath, isNewTable = true, testSchema, partCols = Seq.empty,
+        tableProperties = Map(TableConfig.COLUMN_MAPPING_MODE.getKey -> "name"))
+        .commit(engine, emptyIterable())
+
+      val ex = intercept[IllegalArgumentException] {
+        table.createTransactionBuilder(engine, testEngineInfo, Operation.WRITE)
+          .withTableProperties(
+            engine,
+            Map(TableConfig.COLUMN_MAPPING_MODE.getKey -> "none").asJava)
+          .build(engine)
+      }
+      assert(ex.getMessage.contains("Changing column mapping mode " +
+        "from 'name' to 'none' is not supported"))
+    }
+  }
+
+  test("update column mapping mode from id to name on existing table") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      val table = Table.forPath(engine, tablePath)
+      val schema = new StructType()
+        .add("a", StringType.STRING, true)
+        .add("b", IntegerType.INTEGER, true)
+
+      createTxn(engine, tablePath, isNewTable = true, schema, partCols = Seq.empty,
+        tableProperties = Map(TableConfig.COLUMN_MAPPING_MODE.getKey -> "id"))
+        .commit(engine, emptyIterable())
+
+      val structType = table.getLatestSnapshot(engine).getSchema(engine)
+      assertColumnMappingMetadata(structType.get("a").getMetadata, 1, "UUID")
+      assertColumnMappingMetadata(structType.get("b").getMetadata, 2, "UUID")
+
+      val ex = intercept[IllegalArgumentException] {
+        table.createTransactionBuilder(engine, testEngineInfo, Operation.WRITE)
+          .withTableProperties(
+            engine,
+            Map(TableConfig.COLUMN_MAPPING_MODE.getKey -> "name").asJava)
+          .build(engine)
+          .commit(engine, emptyIterable())
+      }
+      assert(ex.getMessage.contains("Changing column mapping mode " +
+        "from 'id' to 'name' is not supported"))
+    }
+  }
+
+  test("update column mapping mode from name to id on existing table") {
```

Comment:
nit: make the test name tell us a bit more about the expectation of the test: `cannot column mapping mode from name to id on existing table`

Reply:
apply to other test names, too

Reply:
makes sense, done